types/params.go

```diff
@@ -157,7 +154,9 @@ func (params ConsensusParams) Update(params2 *abci.ConsensusParams) ConsensusPar
 		res.Evidence.MaxAge = params2.Evidence.MaxAge
 	}
 	if params2.Validator != nil {
-		res.Validator.PubKeyTypes = params2.Validator.PubKeyTypes
+		// Copy params2.Validator.PubkeyTypes, and set result's value to the copy.
+		// This avoids having to initialize the slice to 0 values, and then write to it again.
+		res.Validator.PubKeyTypes = append([]string{}, params2.Validator.PubKeyTypes...)
```

Comment:
so the simple assignment will result in copy, right? premature optimization?

Reply:
The simple assignment will both reference the same underlying data. Your right, this append may be a premature optimization, but only because golang doesn't have a better copy method! This is actually semi-recommended: https://github.com/go101/go101/wiki/How-to-efficiently-clone-a-slice%3F

Reply:
No need. Comment is enough